jax/interpreters/partial_eval.py

```diff
@@ -187,9 +187,14 @@ def process_call(self, call_primitive, f: lu.WrappedFun, tracers, params):
     lifted_jaxpr = convert_constvars_jaxpr(jaxpr)
     out_tracers = [JaxprTracer(self, pval, None) for pval in out_pvals]
     new_params = dict(params, call_jaxpr=lifted_jaxpr)
+    invars = tuple(it.chain(const_tracers, env_tracers, tracers))
+    if 'donated_invars' in params:
+      new_donated_invars = ((False,) * len(const_tracers) +
```

Comment:
Similarly, in the case where `partial_eval` is used for reverse-mode AD (but only `grad`, not `linearize`,  because that's all about reusing residuals!), it's possible that we could use donation on `const_tracers` because we know they'll never be reused.

Reply:
Indeed as you say this wouldn't work for `linearize` (of `jit`), but it also wouldn't work for `vjp`, because in both cases we might call the linearized function many times and hence can't throw away the residuals after the first execution. It'd only work for `grad`, though of course `grad` is an important special case and might be worth thinking about lightweight ways to provide that optimization.